### PR TITLE
v1.15 Backports 2024-10-02

### PR DESCRIPTION
**Pull request by aanm**
Backporting all of these PRs to avoid backport conflicts and to optimize our CI infrastructure. PRs for the remaining stable branches will happen afterwards.

This will effectively remove Conformance E2E (ci-e2e) from our v1.15 branch.

* [x] #29459
 * [x] #34811 (@aanm) :warning: resolved conflicts
 * [x] #34742 (@aanm) :warning: resolved conflicts
 * [x] #34800 (@aanm) :warning: resolved conflicts
 * [x] #34845 (@aanm)
 * [x] #34843 (@aanm) :warning: resolved conflicts
 * [x] #34847 (@aanm) :warning: resolved conflicts
 * [x] #34848 (@aanm) :warning: resolved conflicts
 * [x] #34866 (@aanm) :warning: resolved conflicts
 * [x] #35088 (@aanm)
 * [x] #35141 (@aanm)
 * [x] #33451  (@aanm)
 * [x] #32528
 * [x] #35227

Once this PR is merged, a GitHub action will update the labels of these PRs:
```upstream-prs
29459 34811 34742 34800 34845 34843 34847 34848 34866 35088 35141 33451 32528 35227
```

### CODE REVIEW

**Comment by aanm** [comment]
/test-backport-1.15

**Comment by aanm** [comment]
/test-backport-1.15

**Comment by aanm** [comment]
/test-backport-1.15

**Review by nebril — APPROVED** [review]

File: .github/workflows/tests-e2e-upgrade.yaml [hunk 1]
```diff
@@ -186,6 +183,7 @@ jobs:
             kpr: 'false'
             tunnel: 'geneve'
             endpoint-routes: 'true'
+            misc: 'socketLB.enabled=false,nodePort.enabled=true,bpf.masquerade=true'
```
Comment (julianwiedmann) [inline .github/workflows/tests-e2e-upgrade.yaml]: This doesn't exist in the `v1.15` version of the conformance-e2e workflow.

File: .github/workflows/tests-e2e-upgrade.yaml [hunk 2]
```diff
@@ -243,6 +241,14 @@ jobs:
             egress-gateway: 'true'
             ingress-controller: 'true'
 
+          - name: '13'
+            # renovate: datasource=docker depName=quay.io/lvh-images/kind
+            kernel: 'rhel8-20240404.144247'
```
Comment (julianwiedmann) [inline .github/workflows/tests-e2e-upgrade.yaml]: In the `v1.15` version of the conformance-e2e workflow this is still `4.19-20240710.064909`.

File: .github/workflows/conformance-e2e.yaml [hunk 3]
```diff
@@ -1,474 +0,0 @@
...
-          # ! NOTE: keep tests-e2e-upgrade.yaml config in sync !
-          # !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
-
-          - name: '1'
-            # renovate: datasource=docker depName=quay.io/lvh-images/kind
-            kernel: '4.19-20240710.064909'
-            kube-proxy: 'iptables'
-            kpr: 'false'
-            tunnel: 'vxlan'
-            host-fw: 'true'
```
Comment (julianwiedmann) [inline .github/workflows/conformance-e2e.yaml:112]: To note - in the `tests-e2e-upgrade` workflow the `host-fw` option is set to false (to work-around downgrade bugs). So if we now solely use that workflow for testing, we're effectively losing coverage.

**Review by julianwiedmann — CHANGES_REQUESTED** [review]
Some minor changes needed.

File: .github/workflows/conformance-e2e.yaml (hunk 3 above)
Comment (aanm) [reply .github/workflows/conformance-e2e.yaml:112]: @julianwiedmann thanks for the review. I've copied the host-fw's specific tests intot he tests-e2e-upgrade tests and skip the upgrade for those configurations. We shouldn't be losing any coverage now.

**Comment by aanm** [comment]
/test-backport-1.15

**Comment by aanm** [comment]
/test-backport-1.15

**Review by joestringer — APPROVED** [review]
I glanced through most of the `.github` files and nothing obvious jumped out at me. Broadly speaking it is very difficult to give a good review on what is effectively a rewrite of significant portions of the workflows on an old branch.

**Review by julianwiedmann — APPROVED** [review]
thank you!

**Comment by aanm** [comment]
/test-backport-1.15

**Comment by aanm** [comment]
/test-backport-1.15

**Pull request merged**